src/track/beatstats.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <QQueue>
+
+// These classes are used to compute statistics descriptors
+// of a series of tempo values and are called from beatutils
+
+class WindowedStatistics {
+
+    QQueue<double> m_window;
+    int m_period;
+    
+    double updateWindow(double newValue) {
+        m_window.enqueue(newValue);
+        if (m_window.count() > m_period) {
+            return m_window.dequeue();
+        } else {
+            return std::nan("");
+        }
+    }
+    virtual double compute() = 0;
+    virtual void update(double newValue, double oldValue) = 0;
+
+  public:
+
+    WindowedStatistics(int period) {
+        assert(period > 0);
+        m_period = period;
+    }
+    virtual ~WindowedStatistics(){};
+
+    double operator()(double newValue) {
+        double oldValue = updateWindow(newValue);
+        update(newValue, oldValue);
+        return compute();
+    }
+    double operator()(void) {
+        return compute();
+    }
+    void operator++(int) {
```

Comment:
Do we really need this? It is probably to much magic.

Reply:
Yeah, we don't need this. I was using it with some test I was doing with a moving average, that I end up not using at all. So this was just a quick hack to increase the period during iteration. I think however that a better API for changing the period might be useful?